### PR TITLE
Add flatpak support

### PR DESCRIPTION
**Pull request by aadcg**
Addresses https://github.com/atlas-engineer/nyxt/issues/3044.

### CODE REVIEW

File: nfiles.lisp [hunk 1]
```diff
@@ -300,14 +300,44 @@ See `expand' for a convenience wrapper."))
...
+(defun flatpak-xdg-config-home (&rest more)
+  "Based on `uiop:xdg-config-home'."
+  (uiop:resolve-absolute-location
+   `(,(or (uiop:getenv-absolute-directory "HOST_XDG_CONFIG_HOME")
+          (uiop:subpathname (user-homedir-pathname) ".config/"))
+     ,more)))
+
 (defmethod resolve ((profile profile) (file config-file))
-  (maybe-xdg #'uiop:xdg-config-home (call-next-method)))
+  #-flatpak
```
Comment (jmercouris) [inline nfiles.lisp:311]: You could just move this branch to line 303 above instead, that way you don't have to put it everywhere.

Comment (aartaka) [reply nfiles.lisp]: Yet another alternative: put them all inside a single `progn` and only use feature expression on this `progn`. That [still counts as top level form](https://cl-community-spec.github.io/pages/Processing-of-Top-Level-Forms.html), while being much easier to include/exclude 😃

**Comment by Ambrevar** [comment]
Looks good to me, please squash before merging.  Thanks!

**Comment by jmercouris** [comment]
Perfect, thank you.

**Pull request merged**